### ISSUE TITLE
Salt 3005.1 still has a failure for a single urlencoded argument not being allowed (i.e. #62624)

### ISSUE THREAD

**Issue by zbukhari-apex** [saltstack/salt#64209]
**Description**
Using salt-api, when a single argument is sent for say a `cmd.run`, it fails. I have to add additional arguments as described in #62624 

3005 should still be supported but looking at #62624 the fix for the problem which is in 3005 / 3005.1 was done in 3006 and not in 3005 afaict.

**Setup**
```
...
external_auth:
  pam:
    saltm1user:
      - .*
      - '@runner'
      - '@wheel'

rest_cherrypy:
  host: 0.0.0.0
  port: 8000
  ssl_key: /etc/salt/saltm1.key
  ssl_crt: /etc/salt/saltm1.crt
...
```

Please be as specific as possible and give set-up details.

- [x] on-prem machine
- [x] VM - vmware on Ubuntu Bionic/Focal
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [x] onedir packaging
- [ ] used bootstrap to install

**Steps to Reproduce the behavior**
```
$         curl -sSk "https://${salt_master}:8000" \
                -b "$cookie_file" \
                -H 'Accept: application/json' \
                -d client=local \
                -d tgt='host*' \
                -d fun='cmd.run' \
                -d arg='whoami'
...

<!DOCTYPE html PUBLIC
"-//W3C//DTD XHTML 1.0 Transitional//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"></meta>
    <title>500 Internal Server Error</title>
    <style type="text/css">
    #powered_by {
        margin-top: 20px;
        border-top: 2px solid black;
        font-style: italic;
    }   

    #traceback {
        color: red;
    }
    </style>
</head>
    <body>
        <h2>500 Internal Server Error</h2>
        <p>The server encountered an unexpected condition which prevented it from fulfilling the request.</p>
        <pre id="traceback"></pre>
    <div id="powered_by">
      <span>
        Powered by <a href="http://www.cherrypy.org">CherryPy 18.6.1</a>
      </span>
    </div>
    </body>
</html>
```

**Expected behavior**
```
{"return": [{"host1.dom.tld": "root", "host2.dom.tld": "root"}]}
```

For it to work in 3005.1 I have to pass in another argument as mentioned in 62624.
```
$         curl -sSk "https://${salt_master}:8000" \
                -b "$cookie_file" \
                -H 'Accept: application/json' \
                -d client=local \
                -d tgt='host*' \
                -d fun='cmd.run' \
                -d arg='whoami' \
                -d arg='c=d'
```

**Screenshots**
N/A

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
Salt Version:
          Salt: 3005.1
 
Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.0
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
        Python: 3.9.16 (main, Jan  6 2023, 22:49:49)
  python-gnupg: 0.4.8
        PyYAML: 5.4.1
         PyZMQ: 23.2.0
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: ubuntu 18.04 bionic
        locale: utf-8
       machine: x86_64
       release: 5.4.0-139-generic
        system: Linux
       version: Ubuntu 18.04 bionic

```
</details>

**Additional context**

**Comment by OrangeDog** [comment]
"Supported" does not mean that bug fixes are back-ported and released, unfortunately.

3005.2 might get released if a new CVE is identified, but it's unlikely to have this fix in it either.

**Comment by whytewolf** [comment]
3006.1 is set to be release very soon. we moved up the schedule because of the jinja issues being so critical.

**Comment by OrangeDog** [comment]
> highstate from the salt master never seems to return

#62881

**Comment by zbukhari-apex** [comment]
Awesome news! Thanks for the bug reference.

Not all heroes wear capes! Unless Saltstack employees are donning capes ;-) :-D

Happy Friday!

**Comment by whytewolf** [comment]
@zbukhari-apex just a heads up, we just released 3006.1

**Comment by zbukhari-apex** [comment]
I saw. Planning on testing it out either this week or next week. Thank you!
